File: Allin.Backend.Net/Microservices/Admin/Allin.Admin.Application/Models/Roles/RoleModel.cs
using Allin.Admin.Application.Common;
using Allin.Admin.Infrastructure.Persistence;
using Allin.Common.Utilities.Mappings;

namespace Allin.Admin.Application.Models
{
    public class RoleModel : AdminBaseModel, IMapFrom<Role, RoleModel>
    {
        public long? ParentId { get; set; }
        public string Title { get; set; }
        public string UniqueName { get; set; }
        public string Description { get; set; }
        // public IList<string> Phones { get; set; }
        public IEnumerable<PermissionModel> Permissions { get; set; }
        // public virtual ICollection<PermissionModel> UserRoles { get; set; } = new List<UserRole>();
    }
}

File: allin-front-angular/src/app/subsystems/admin/models/queries/role-model.ts
import { TreeNode } from "primeng/api";
import { PermissiontModel } from "./permission-model";

export interface RoleModel {
    id: number;
    title: string;
    uniqueName: string;
    parentId?: number;
    description: string;
    permissions: TreeNode<PermissiontModel>[];


}